bin/style

```diff
@@ -10,104 +8,136 @@
 # Otherwise any uncommitted source files are linted. If there is no uncommitted change
 # then the files in the most recent commit are linted.
 #
-set all no
-set git_clang_format no
-set c_files
-set files
+
+# Allow for handling filenames with spaces
+IFS=$'\n'
+
+typeset all=no
+typeset git_clang_format=no
+typeset -a c_files=()
+typeset -a files=()
 
 # Deal with any CLI flags.
-while set -q argv[1]
-    if test "$argv[1]" = "--all"
-    or test "$argv[1]" = "all"
-        set all yes
-        set -e argv[1]
-    else
-        break
-    end
-end
+while [[ ${1} != "" ]]
+do
+    case ${1} in
+        --all | all )
+            all=yes
+            ;;
+        * )
+            break
+            ;;
+    esac
+    shift
+done
 
-if test $all = yes
-and set -q argv[1]
-    echo "Unexpected arguments: '$argv'" >&2
+if [[ ${all} == yes && ${1} != "" ]]
+then
+    echo "Unexpected arguments: '${1}'" >&2
     exit 1
-end
+fi
 
-if test $all = yes
-    set files (git status --porcelain --short --untracked-files=all | sed -e 's/^ *[^ ]* *//')
-    if set -q files[1]
+if [[ ${all} == yes ]]
+then
+    files=( $(git status --porcelain --short --untracked-files=all | sed -e 's/^ *[^ ]* *//') )
+
+    if [[ ${files[0]} != "" ]]
+    then
         echo >&2
-        echo You have uncommited changes. Cowardly refusing to restyle the entire code base. >&2
+        echo You have uncommitted changes. Cowardly refusing to restyle the entire code base. >&2
         echo >&2
-        exit 1
-    end
-    set files src/**.h src/**.c
-else if set -q argv[1]
-    set files
-    for f in $argv
-        if test -f $f
-            set files $files $f
-        else if test -d $f
-            set files $files {$f}**.c {$f}**.h
-        end
-    end
+        exit
+    fi
+
+    files=( $(find src -name "**.h" -o -name "**.c") )
+elif [[ ${1} != "" ]]
+then
+    for next_file in $*
```

Comment:
Nitpick: This can be shortened to `for next_file` but I think it's better to explicitly specifiy `$*`.